python/plugins/processing/algs/qgis/RasterCalculator.py

```diff
@@ -125,15 +127,16 @@ def processAlgorithm(self, parameters, context, feedback):
             layersDict = {os.path.basename(lyr.source().split(".")[0]): lyr for lyr in layers}
```

Comment:
This doesn't work when you are using multiple unnamed intermediary algorithms (*Model 2*). Each one has `OUTPUT` as the result of `os.path.basename(lyr.source().split(".")[0])` which means that only the final layer to be read makes it into the dictionary as the others will be overwritten.

Reply:
Is there any reason why we shouldn't use `layersDict = {lyr.source(): lyr for lyr in layers}` here? Tested and works with *Model 2* (in conjunction with the change suggested at L284). Not sure if there could be os-related issues (I'm testing on Windows).

Reply:
applied and your suggestion works correctly